modules/gapi/test/common/gapi_tests_common.hpp

```diff
@@ -155,20 +155,60 @@ class TestFunctional
         }
     }
 
+    void initMatsFromImages(int channels, const std::string& fileNamePattern, int format)
+    {
+        auto getDataPath = [](const std::string& relativePath)
+        {
+            return perf::TestBase::getDataPath(relativePath);
+        };
+        initTestDataPath();
+        switch (channels)
+        {
+            case 1:
+                in_mat1 = cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format)),
+                                     cv::IMREAD_GRAYSCALE);
+                in_mat2 = cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format + 1)),
+                                     cv::IMREAD_GRAYSCALE);
+                break;
+            case 3:
+                in_mat1 = cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format)));
+                in_mat2 = cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format + 1)));
+                break;
+            case 4:
+                cvtColor(cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format))),
+                         in_mat1, cv::COLOR_BGR2BGRA);
+                cvtColor(cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format + 1))),
+                         in_mat2, cv::COLOR_BGR2BGRA);
+                break;
+            default:
+                FAIL() << "Unexpected number of channels: " << channels;
+        }
```

Comment:
This code can be simpler:
```c++
GAPI_Assert(channels == 1 || channels == 3 || channels == 4);
const int flags = (channels == 1) ? cv::IMREAD_GRAYSCALE : cv::IMREAD_COLOR;
cv::Mat m1 = cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format);
cv::Mat m2 = cv::imread(getDataPath(cv::format(fileNamePattern.c_str(), format + 1);
if (channels == 4) {
    cvtColor(m1, in_mat1, cv::COLOR_BGR2BGRA);
    cvtColor(m2, in_mat2, cv::COLOR_BGR2BGRA);
} else {
    std::tie(in_mat1, in_mat2) = std::make_tuple(m1, m2);
}
```

BTW why do we need this `format + 1` thing?

Reply:
Agree with the simplification, it's nice. Thanks!
Formatting is convenient here: it is supposed to that we are loading two files with the same names but the number in them like "optFlow01.jpg" and "optFlow02.jpg".
Ex., two instances of the test can take `fileNamePattern` = "cv/optflow/rock_%01d.bmp" and `format` = 1, `format` = 2. Processing the first instance, we initialize `in_mat1` and `in_mat2` with "cv/optflow/rock_1.bmp" and "cv/optflow/rock_2.bmp". Processing the second one, they are "cv/optflow/rock_2.bmp" and "cv/optflow/rock_3.bmp".
It can be useful

Reply:
Ok, I see it now.